modules/transport-netty4/src/test/java/org/elasticsearch/transport/netty4/SimpleNetty4TransportTests.java

```diff
@@ -79,7 +77,15 @@ public void executeHandshake(
                 if (doHandshake) {
                     super.executeHandshake(node, channel, profile, listener);
                 } else {
-                    listener.onResponse(TransportVersionUtils.minimumCompatibilityVersion(version));
+                    TransportVersion version = getVersion();
```

Comment:
why we are not using `TransportVersionUtils.minimumCompatibilityVersion` anymore?
also I noticed the same code in [SimpleSecurityNetty4ServerTransportTests.java](https://github.com/elastic/elasticsearch/pull/94892/files#diff-2a95a1d96fb86fe9208bf60f97a0e438532dfed952c65c761c941d7f1c5ce1a8)
perhaps it deserves a util method?

Reply:
actually not a util method, but perhaps a method on TransportVersion. It might be just returning a TransportVersion.MINIMUM_COMPATIBLE but would be a good place to keep the javadoc on what minimum compatible is/why it is using that constant

Reply:
Then we could just put javadoc on the field itself? How transportversion is used & what it means will be in the javadoc for TransportVersion (once we've finalised how that will work)

Reply:
agree we can create the javadoc later, once we have all discussions finalised 
how about having `TransportVersionUtils.minimumCompatibilityVersion` or `TransportVersion.minimumCompatibilityVersion` method with this code block? It is used in two places already

Reply:
I'm not sure what benefit there is adding a method that just returns the static field?

Reply:
I meant the full block of code that is duplicated 
```
 TransportVersion version = getVersion();
                TransportVersion compatible;
                if (version.equals(TransportVersion.CURRENT)) {
                    compatible = TransportVersion.MINIMUM_COMPATIBLE;
                } else {
                    throw new IllegalArgumentException("Unknown transport version " + getVersion());
                }
```

Reply:
Ah, I see what you mean. Could do; I've had another look at it, and simplified the check to just an assert (as it looks like only CURRENT is ever used here)